packages/fhir-converter/src/templates/cda/Header.hbs

```diff
@@ -1,74 +1,76 @@
-{{#with msg.ClinicalDocument}}
-    {{>Resources/Composition.hbs composition=this ID=(generateUUID (toJsonString this))}},
+{{#with msg.ClinicalDocument as |document|}}
+    {{>Resources/Composition.hbs composition=document ID=(generateUUID (toJsonString document))}},
     {{#with (evaluate 'Utils/GeneratePatientId.hbs' obj=@metriportPatientId) as |patientId|}}
-    	{{>References/Composition/subject.hbs ID=(generateUUID (toJsonString ../this)) REF=(concat 'Patient/' patientId.Id)}},
+    	{{>References/Composition/subject.hbs ID=(generateUUID (toJsonString document)) REF=(concat 'Patient/' patientId.Id)}},
     {{/with}}
 
-        {{#if this.legalAuthenticator}}
-            {{#with (evaluate 'Utils/GeneratePractitionerId.hbs' obj=this.legalAuthenticator.assignedEntity) as |practitionerId|}}
-                {{>Resources/Practitioner.hbs practitioner=../legalAuthenticator.assignedEntity ID=practitionerId.Id}},
+    {{#if document.legalAuthenticator}}
+        {{#with (evaluate 'Utils/GeneratePractitionerId.hbs' obj=document.legalAuthenticator.assignedEntity) as |practitionerId|}}
+            {{>Resources/Practitioner.hbs practitioner=document.legalAuthenticator.assignedEntity ID=practitionerId.Id}},
+        {{/with}}
+    {{/if}}
+
+    {{#if document.componentOf.encompassingEncounter}}
+        {{!-- do we want to include this? Kinda a duplicate of more rich data later in encounter section --}}
+        {{>Resources/Encounter.hbs encounter=document.componentOf.encompassingEncounter ID=(generateUUID (toJsonString document.componentOf.encompassingEncounter))}},
```

Comment:
this encounter we currently include is kinda like a poor mans version of encounter later on in document